bin/lint-markdown-api-history.ts

```diff
@@ -0,0 +1,616 @@
+#!/usr/bin/env node
+
+import { readFile, access, constants } from 'node:fs/promises';
+import { resolve } from 'node:path';
+import * as minimist from 'minimist';
+
+import { DocsWorkspace } from '../lib/markdown';
+import { URI } from 'vscode-uri';
+import { parse as parseYaml } from 'yaml';
+
+import type { HTML, Heading } from 'mdast';
+import type { Node, Literal } from 'unist';
+import type { visit as VisitFunction } from 'unist-util-visit';
+import type { fromHtml as FromHtmlFunction } from 'hast-util-from-html';
+import type { fromMarkdown as FromMarkdownFunction } from 'mdast-util-from-markdown';
+import { dynamicImport } from '../lib/helpers';
+import Ajv, { JSONSchemaType, ValidateFunction } from 'ajv';
+import AdmZip = require('adm-zip');
+
+// "<any char>: <match group>"
+const possibleStringRegex = /^[ \S]+?: *?(\S[ \S]+?)$/gm;
+const nonAlphaNumericDotRegex = /[^a-zA-Z0-9.]/g;
+
+interface ChangeSchema {
+  'pr-url': string;
+  'breaking-changes-header'?: string;
+  description?: string;
+}
+
+interface ApiHistory {
+  added?: ChangeSchema[];
+  deprecated?: ChangeSchema[];
+  changes?: ChangeSchema[];
+}
+
+interface Options {
+  checkPlacement: boolean;
+  checkPullRequestLinks: boolean;
+  breakingChangesFile: string;
+  checkStrings: boolean;
+  ignoreGlobs: string[];
+  schema: string;
+}
+
+interface HTMLWithPreviousNode extends HTML {
+  previousNode?: Node;
+}
+
+// If you change this, you might want to update the one in the website transformer
+export interface PrReleaseVersions {
+  release: string | null;
+  backports: Array<string>;
+}
+
+// If you change this, you might want to update the one in the website transformer
+export type PrReleaseVersionsContainer = { [key: number]: PrReleaseVersions };
+
+// If you change this, you might want to update the one in the website transformer
+interface PrReleaseArtifact {
+  data: PrReleaseVersionsContainer;
+  endCursor: string;
+}
+
+function getCIPrNumber(): number | null {
+  if (process.env.GITHUB_REF_NAME?.endsWith('/merge')) {
+    // https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+    return Number(process.env.GITHUB_REF_NAME.split('/')[0]);
+  } else if (process.env.CIRCLE_PULL_REQUEST?.includes('/pull/')) {
+    // https://circleci.com/docs/variables/#built-in-environment-variables
+    return Number(process.env.CIRCLE_PULL_REQUEST.split('/').at(-1));
+  } else {
+    return null;
+  }
+}
+
+let _allPrReleaseVersions: PrReleaseVersionsContainer;
+
+// If you change this, you might want to update the one in the website transformer
+// TODO: Change this when GH_TOKEN isn't needed to fetch PR release versions anymore
+async function getAllPrReleaseVersions(): Promise<PrReleaseVersionsContainer> {
+  try {
+    if (_allPrReleaseVersions) {
+      return _allPrReleaseVersions;
+    }
+
+    if (process.env.NODE_ENV === 'test') {
```

Comment:
Are these pre-loaded for testing? Instead of returning these in the method itself, could we create a mock in the test environments?

Reply:
Agreed here. Spying on the function within the test file to return the fixture data sounds like a better approach.